yalafi/packages/cleveref.py

```diff
@@ -0,0 +1,153 @@
+
+#
+#   YaLafi module for LaTeX package cleveref
+#
+
+from yalafi.defs import InitModule, Macro, VoidToken
+from yalafi import utils
+import re
+import sys
+
+
+require_packages = []
+
+re_ref = re.compile(r's/\\(\\cref|\\Cref)\{([^\}\{]+)\}/(.*)/g')
+re_ref_range = re.compile(r's/\\(\\crefrange|\\Crefrange)\{([^\}\{]+)\}\{([^\}\{]+)\}/(.*)/g')
+re_command = re.compile(r'''
+s/\\
+(                   # group 1: command
+    \\
+    (               # group 2: capturing cC
+        \[cC\]
+    )?
+    [a-zA-Z@]+
+)
+(                   # group 3: possible arguments
+    (?:\{\.\*\})+
+)?
+\s*                 # possible whitespaces inbetween
+/
+    (.*)            # group 4: replacement
+/g
+''', re.VERBOSE)
+re_cC = re.compile(r'(\[cC\])')
+re_dot = re.compile(r'\\\.')
+re_double_backslash = re.compile(r'\\\\')
+
+macro_read_sed = '\\LTReadSed'
+
+msg_poorman_option = f'''
+*** LaTeX warning:
+*** To use cleveref with YaLafi, you need to use the 'poorman' option
+*** and {macro_read_sed} to load the .sed file.
+'''
+
+msg_sed_not_loaded = f'''To use cleveref with YaLafi, you should use {macro_read_sed} to load
+*** the sed file, e.g. '{macro_read_sed}{{main.sed}}' if your LaTeX
+*** document is called 'main.tex'.
+'''
+
+
+def init_module(parser, options):
+    parms = parser.parms
+    parms.newcommand_ignore.append(macro_read_sed)
+
+    macros_latex = r'''
+        \newcommand{\crefname}[3]{}
+        \newcommand{\Crefname}[3]{}
+        \newcommand{\nobreakspace}{~}
+    '''
+
+    macros_python = [
+
+        Macro(parms, macro_read_sed, args='A', repl=h_read_sed),
+        Macro(parms, '\\label', args='OA', repl=''),
+        Macro(parms, '\\cref', args='A', repl=h_cref_warning),
+        Macro(parms, '\\Cref', args='A', repl=h_cref_warning),
+        Macro(parms, '\\crefrange', args='AA', repl=h_cref_warning),
+        Macro(parms, '\\Crefrange', args='AA', repl=h_cref_warning),
+
+    ]
+
+    environments = []
+
+    poorman_warning(options)

```

Comment:
I want to show a warning if the package is loaded without the `poorman` option. Currently, I just use `sys.stderr.write`. I would like it better, if I could use something like `utils.latex_error` and display the warning also in the output where `\usepackage{cleveref}` is loaded. Is there an easy possibility to do that?